src/ast/passes/fold_literals.cpp

```diff
@@ -29,6 +29,8 @@ class LiteralFolder : public Visitor<LiteralFolder, std::optional<Expression>> {
   std::optional<Expression> visit(PositionalParameter &param);
   std::optional<Expression> visit(Call &call);
   std::optional<Expression> visit(Expression &expr);
+  // N.B. this should never return anything but a std::nullopt
```

Comment:
I would just drop this comment. It's the kind that will become quickly out of date and was more related to your original motivation than providing a good context clue for the reader.

Reply:
Done.